src/app/app.component.ts

```diff
@@ -198,51 +204,91 @@ export class AppComponent implements AfterViewInit {
+    await this.lfRepositoryBrowser?.nativeElement.initAsync(this.lfRepoTreeNodeService, focusedNode as LfRepoTreeNode);
   }
 
+  isNodeSelectable = (node: LfRepoTreeNode) => {
+    if (node.entryType == EntryType.Folder) {
+      return true;
+    }
+    else {
+      return false;
+    }
+  };
+
   get isLoggedIn(): boolean {
     return this.loginComponent?.nativeElement?.state === LoginState.LoggedIn;
   }
 
   // Tree event handler methods
-  async onOkClick(okClickEvent: Event) {
-    const selectedNode = (okClickEvent as CustomEvent<TreeNode>).detail;
-    const breadcrumbs = this.lfFolderBrowserElement?.nativeElement?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
-    const path = selectedNode.path;
-    this.selectedFolder = {
-      entryId,
-      path,
-      displayName: this.getFolderNameText(entryId, path),
-      displayPath: this.getFolderPathTooltip(path)
-    };
-    if (breadcrumbs) {
-      this.selectedFolder.breadcrumbs = breadcrumbs;
+  async onSelectFolder() {
+    const selectedNode = this.lfRepositoryBrowser.nativeElement.currentFolder as LfRepoTreeNode;
+    let entryId = Number.parseInt(selectedNode.id, 10);
+    const selectedFolderPath = selectedNode.path;
+    if (selectedNode.entryType == EntryType.Shortcut) {
+      entryId = selectedNode.targetId;
     }
-    const nodeId = selectedNode.id;
     const repoId = (await this.repoClient.getCurrentRepoId());
     const waUrl = this.loginComponent.nativeElement.account_endpoints.webClientUrl;
-    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId, waUrl, selectedNode.isContainer);
     this.expandFolderBrowser = false;
+    this.lfSelectedFolder = {
+      selectedNodeUrl: getEntryWebAccessUrl(entryId.toString(), repoId, waUrl, selectedNode.isContainer),
+      selectedFolderName: this.getFolderNameText(entryId, selectedFolderPath),
+      selectedFolderPath: selectedFolderPath
+    }
+  }
+
+  get shouldShowSelect(): boolean {
+    return !this.shouldShowOpen && !!this.lfRepositoryBrowser?.nativeElement?.currentFolder;
+  }
+
+  get shouldShowOpen(): boolean {
+    return !!this.entrySelected;
+  }
+
+  onEntrySelected(event) {
+    const customEvent = event as CustomEvent<LfTreeNode[]>;
+    const treeNodesSelected: LfTreeNode[] = customEvent.detail;
+    this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
+  }
+
+  async onOpenNode() {
+    await this.lfRepositoryBrowser?.nativeElement?.openSelectedNodesAsync();
   }
 
   async onClickBrowse() {
     this.expandFolderBrowser = true;
     await this.initializeTreeAsync();
   }
 
-  private getFolderPathTooltip(path: string): string {
+  get selectedFolderDisplayName(): string {
     const FOLDER_BROWSER_PLACEHOLDER = this.localizationService.getString('FOLDER_BROWSER_PLACEHOLDER');
-    return path ? PathUtils.createDisplayPath(path) : FOLDER_BROWSER_PLACEHOLDER;
+    return this.lfSelectedFolder?.selectedFolderName ?? FOLDER_BROWSER_PLACEHOLDER;
```

Comment:
is this actuall used?, looks like the getter is used in the html, but not the setter